formats/json/commonMain/src/kotlinx/serialization/json/internal/Polymorphic.kt

```diff
@@ -79,3 +75,8 @@ private fun throwSerializerNotFound(type: String?, jsonTree: JsonObject): Nothin
         else "class discriminator '$type'"
     throw JsonDecodingException(-1, "Polymorphic serializer was not found for $suffix", jsonTree.toString())
 }
+
+internal fun SerialDescriptor.classDiscriminator(json: Json): String =
+    annotations.filterIsInstance<JsonClassDiscriminator>().singleOrNull()?.discriminator
```

Comment:
Iterating over annotations for each annotated class can slow down deserialization, it might be worth caching the discriminator.

Reply:
For most of the classes, this array will contain 0 or 1 element (because we do not have a lot of serial info class-level annotations). So I assume this linear lookup will be faster than two (SerialDescriptor -> DescriptorSchemaCache.Key) hash map accesses, or at least not slower